modules/swagger-codegen/src/main/resources/Java/apiInvoker.mustache

```diff
@@ -25,10 +25,30 @@ public class ApiInvoker {
   {{! wikia }}
   private static final String RESTRICTED_HEADERS_PROPERTY = "sun.net.http.allowRestrictedHeaders";
   {{! /wikia }}
+  public interface ClientFactory {
+    public Client getClient(String path);
+  }
+
+  public class DefaultClientFactory implements ClientFactory {
+    private Map<String, Client> hostMap = new HashMap<String, Client>();
+
+    protected Client createClient(String path) {
+      return ClientBuilder.newClient();
+    }
+
+    public Client getClient(String path) {
```

Comment:
Ah yeah.. maybe `baseUrl` or `url` makes more sense? Either way, it should be consistent :)